public class cn.trinea.android.common.util.SerializeUtils extends java.lang.Object
{

    public void <init>()
    {
        cn.trinea.android.common.util.SerializeUtils r0;

        r0 := @this: cn.trinea.android.common.util.SerializeUtils;

        specialinvoke r0.<java.lang.Object: void <init>()>();

        return;
    }

    public static java.lang.Object deserialization(java.lang.String)
    {
        java.lang.Throwable $r5, $r7, $r8, $r10, $r11;
        java.io.ObjectInputStream $r2, $r3;
        java.io.FileInputStream $r4;
        java.lang.RuntimeException $r6;
        java.lang.Object $r1;
        java.lang.ClassNotFoundException $r9;
        java.lang.String $r0;

        $r0 := @parameter0: java.lang.String;

        $r2 = null;

     label01:
        $r3 = new java.io.ObjectInputStream;

        $r4 = new java.io.FileInputStream;

     label02:
        specialinvoke $r4.<java.io.FileInputStream: void <init>(java.lang.String)>($r0);

        specialinvoke $r3.<java.io.ObjectInputStream: void <init>(java.io.InputStream)>($r4);

     label03:
        $r1 = virtualinvoke $r3.<java.io.ObjectInputStream: java.lang.Object readObject()>();

        virtualinvoke $r3.<java.io.ObjectInputStream: void close()>();

     label04:
        virtualinvoke $r3.<java.io.ObjectInputStream: void close()>();

     label05:
        return $r1;

     label06:
        $r5 := @caughtexception;

        $r6 = new java.lang.RuntimeException;

        specialinvoke $r6.<java.lang.RuntimeException: void <init>(java.lang.String,java.lang.Throwable)>("IOException occurred. ", $r5);

        throw $r6;

     label07:
        $r7 := @caughtexception;

        $r3 = null;

     label08:
        $r6 = new java.lang.RuntimeException;

        specialinvoke $r6.<java.lang.RuntimeException: void <init>(java.lang.String,java.lang.Throwable)>("FileNotFoundException occurred. ", $r7);

        throw $r6;

     label09:
        $r8 := @caughtexception;

        $r2 = $r3;

     label10:
        if $r2 == null goto label12;

     label11:
        virtualinvoke $r2.<java.io.ObjectInputStream: void close()>();

     label12:
        throw $r8;

     label13:
        $r9 := @caughtexception;

     label14:
        $r6 = new java.lang.RuntimeException;

        specialinvoke $r6.<java.lang.RuntimeException: void <init>(java.lang.String,java.lang.Throwable)>("ClassNotFoundException occurred. ", $r9);

        throw $r6;

     label15:
        $r8 := @caughtexception;

        goto label10;

     label16:
        $r10 := @caughtexception;

     label17:
        $r6 = new java.lang.RuntimeException;

        specialinvoke $r6.<java.lang.RuntimeException: void <init>(java.lang.String,java.lang.Throwable)>("IOException occurred. ", $r10);

        throw $r6;

     label18:
        $r11 := @caughtexception;

        $r6 = new java.lang.RuntimeException;

        specialinvoke $r6.<java.lang.RuntimeException: void <init>(java.lang.String,java.lang.Throwable)>("IOException occurred. ", $r11);

        throw $r6;

     label19:
        $r10 := @caughtexception;

        $r2 = $r3;

        goto label17;

     label20:
        $r9 := @caughtexception;

        $r2 = $r3;

        goto label14;

     label21:
        $r7 := @caughtexception;

        goto label08;

        catch java.io.FileNotFoundException from label02 to label03 with label07;
        catch java.lang.ClassNotFoundException from label02 to label03 with label13;
        catch java.io.IOException from label02 to label03 with label16;
        catch java.lang.Throwable from label01 to label03 with label15;
        catch java.io.FileNotFoundException from label03 to label04 with label21;
        catch java.lang.ClassNotFoundException from label03 to label04 with label20;
        catch java.io.IOException from label03 to label04 with label19;
        catch java.lang.Throwable from label03 to label04 with label09;
        catch java.io.IOException from label04 to label05 with label06;
        catch java.lang.Throwable from label08 to label09 with label09;
        catch java.io.IOException from label11 to label12 with label18;
        catch java.lang.Throwable from label14 to label15 with label15;
        catch java.lang.Throwable from label17 to label18 with label15;
    }

    public static void serialization(java.lang.String, java.lang.Object)
    {
        java.lang.Throwable $r5, $r7, $r8, $r9, $r10;
        java.io.FileOutputStream $r4;
        java.lang.RuntimeException $r6;
        java.lang.Object $r1;
        java.io.ObjectOutputStream $r2, $r3;
        java.lang.String $r0;

        $r0 := @parameter0: java.lang.String;

        $r1 := @parameter1: java.lang.Object;

        $r2 = null;

     label01:
        $r3 = new java.io.ObjectOutputStream;

        $r4 = new java.io.FileOutputStream;

     label02:
        specialinvoke $r4.<java.io.FileOutputStream: void <init>(java.lang.String)>($r0);

        specialinvoke $r3.<java.io.ObjectOutputStream: void <init>(java.io.OutputStream)>($r4);

     label03:
        virtualinvoke $r3.<java.io.ObjectOutputStream: void writeObject(java.lang.Object)>($r1);

        virtualinvoke $r3.<java.io.ObjectOutputStream: void close()>();

     label04:
        virtualinvoke $r3.<java.io.ObjectOutputStream: void close()>();

     label05:
        return;

     label06:
        $r5 := @caughtexception;

        $r3 = null;

     label07:
        $r6 = new java.lang.RuntimeException;

        specialinvoke $r6.<java.lang.RuntimeException: void <init>(java.lang.String,java.lang.Throwable)>("FileNotFoundException occurred. ", $r5);

        throw $r6;

     label08:
        $r7 := @caughtexception;

        $r2 = $r3;

     label09:
        if $r2 == null goto label11;

     label10:
        virtualinvoke $r2.<java.io.ObjectOutputStream: void close()>();

     label11:
        throw $r7;

     label12:
        $r8 := @caughtexception;

     label13:
        $r6 = new java.lang.RuntimeException;

        specialinvoke $r6.<java.lang.RuntimeException: void <init>(java.lang.String,java.lang.Throwable)>("IOException occurred. ", $r8);

        throw $r6;

     label14:
        $r7 := @caughtexception;

        goto label09;

     label15:
        $r9 := @caughtexception;

        $r6 = new java.lang.RuntimeException;

        specialinvoke $r6.<java.lang.RuntimeException: void <init>(java.lang.String,java.lang.Throwable)>("IOException occurred. ", $r9);

        throw $r6;

     label16:
        $r10 := @caughtexception;

        $r6 = new java.lang.RuntimeException;

        specialinvoke $r6.<java.lang.RuntimeException: void <init>(java.lang.String,java.lang.Throwable)>("IOException occurred. ", $r10);

        throw $r6;

     label17:
        $r8 := @caughtexception;

        $r2 = $r3;

        goto label13;

     label18:
        $r5 := @caughtexception;

        goto label07;

        catch java.io.FileNotFoundException from label02 to label03 with label06;
        catch java.io.IOException from label02 to label03 with label12;
        catch java.lang.Throwable from label01 to label03 with label14;
        catch java.io.FileNotFoundException from label03 to label04 with label18;
        catch java.io.IOException from label03 to label04 with label17;
        catch java.lang.Throwable from label03 to label04 with label08;
        catch java.io.IOException from label04 to label05 with label16;
        catch java.lang.Throwable from label07 to label08 with label08;
        catch java.io.IOException from label10 to label11 with label15;
        catch java.lang.Throwable from label13 to label14 with label14;
    }
}
